src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3794,6 +3756,11 @@ bool os::Linux::setup_large_page_type(size_t page_size) {
 }
 
 void os::large_page_init() {
+  size_t default_large_page_size = scan_default_large_page_size();
+  os::Linux::_default_large_page_size = default_large_page_size;
```

Comment:
I would move this below the checking if large pages are enabled. I'm also not sure if this refactoring made the setup easier to follow. I would have preferred the old way, but if you and Thomas prefer this I'm ok with it.

Reply:
I don't have strong feelings either way but I think the new refactoring is slightly clearer.